packages/components/src/mobile/global-styles-context/utils.native.js

```diff
@@ -286,6 +309,52 @@ function normalizeFontSizes( fontSizes ) {
 	return normalizedFontSizes;
 }
 
+export function useMobileGlobalStylesColors( type = 'colors' ) {
+	const colorGradientSettings = useMultipleOriginColorsAndGradients();
+	const availableThemeColors = colorGradientSettings?.[ type ]?.reduce(
+		( colors, origin ) => colors.concat( origin?.[ type ] ),
+		[]
+	);
+	// Default editor colors/gradients if it's not a block-based theme.
+	const colorPalette =
+		type === 'colors' ? 'color.palette' : 'color.gradients';
+	const editorDefaultPalette = useSetting( colorPalette );
+
+	return availableThemeColors.length >= 1
+		? availableThemeColors
+		: editorDefaultPalette;
+}
+
+export function getColorsAndGradients(
+	defaultEditorColors,
+	defaultEditorGradients,
+	rawFeatures
+) {
+	const features = rawFeatures ? JSON.parse( rawFeatures ) : {};
+
+	return {
+		__experimentalGlobalStylesBaseStyles: null,
+		__experimentalFeatures: {
+			color: {
+				...( ! features?.color
+					? {
+							text: true,
+							background: true,
+							palette: {
+								default: defaultEditorColors,
+							},
+							gradients: {
+								default: defaultEditorGradients,
+							},
+					  }
+					: features?.color ),
+				defaultPalette: !! defaultEditorColors,
+				defaultGradients: !! defaultEditorGradients,
```

Comment:
I noticed that `defaultEditorColors` and `defaultEditorGradients` values are fetched from `settings.colors` and `settings.gradients` in:

https://github.com/WordPress/gutenberg/blob/9eb0f97ea0c37b3cdda39941081eacc92e5fae15/packages/editor/src/components/provider/index.native.js#L366-L367

Note that if they are not defined an empty array is provided, so I'm wondering if to determine the value of `defaultPalette` and `defaultGradients`, we should also check if the array is empty ,wdyt?

Reply:
Nice suggestion! I updated in https://github.com/WordPress/gutenberg/pull/38417/commits/94ab4c38d4fae40b13abc80e14c10d187b7ba358